src/azure-cli-core/azure/cli/core/commands/arm.py

```diff
@@ -276,7 +280,15 @@ def handler(args):
         # Done... update the instance!
         getterargs[setter_arg_name] = parent if child_collection_prop_name else instance
         setter = get_op_handler(setter_op)
+        no_wait = expose_no_wait and setterargs.get('raw', None)
```

Comment:
I made the change. Please take a look. It is easy to rollback if we feel the complexity not worth it